src/gcodeExport.cpp

```diff
@@ -57,7 +57,37 @@ void GCodeExport::preSetup(const MeshGroup* meshgroup)
 
     for (unsigned int extruder_nr = 0; extruder_nr < extruder_count; extruder_nr++)
     {
-        const ExtruderTrain* train = meshgroup->getExtruderTrain(extruder_nr);
+        ExtruderTrain* train = meshgroup->getExtruderTrain(extruder_nr);
+
+        // Cura only has the global stack when it is a single-extrusion machine.
+        // In this case, wee need to update the extruder train settings according to the global stack.
+        if (extruder_count == 1 and extruder_nr == 0)
+        {
+            // Bellow is a list of settings that we need to update according to the global stack.
+            // This list is almost the same as the code block below except that some settings are
+            // commented out. This is because those ones are ONLY applicable to multi-extrusion machines.
+            std::vector<std::string> settings_to_fetch_list;
+            settings_to_fetch_list.push_back("material_diameter");
+            settings_to_fetch_list.push_back("extruder_prime_pos_x");
+            settings_to_fetch_list.push_back("extruder_prime_pos_y");
+            settings_to_fetch_list.push_back("extruder_prime_pos_z");
+            settings_to_fetch_list.push_back("extruder_prime_pos_abs");
+            settings_to_fetch_list.push_back("machine_nozzle_temp_enabled");
+            settings_to_fetch_list.push_back("prime_blob_enable");
+            settings_to_fetch_list.push_back("machine_nozzle_size");
+            //settings_to_fetch_list.push_back("machine_nozzle_offset_x");  // only for multi-extrusion machines
+            //settings_to_fetch_list.push_back("machine_nozzle_offset_y");  // only for multi-extrusion machines
+            settings_to_fetch_list.push_back("material_guid");
+            //settings_to_fetch_list.push_back("machine_extruder_start_code");  // only for multi-extrusion machines
+            //settings_to_fetch_list.push_back("machine_extruder_end_code");  // only for multi-extrusion machines
+            settings_to_fetch_list.push_back("retraction_prime_speed");
```

Comment:
Shouldn't it need to just update ALL settings in the extruder train?

If we're debugging later for some bug in other pieces of code, we'll discover that somehow the `retraction_prime_speed` gets inherited from the global stack into the extruder stack, but its sister `retraction_retract_speed` doesn't. I'd expect the extruder train to contain all settings properly, not just the ones that happened to be necessary on line 103 of GCodeExport. And I'd expect the extruder train to contain all settings from the beginning of the slicing, not just from the Export stage onwards.